app/javascript/dashboard/routes/dashboard/settings/macros/MacroEditor.vue

```diff
@@ -0,0 +1,599 @@
+          });
+        default:
+          return undefined;
+      }
+    },
+    showActionInput(actionName) {
+      if (actionName === 'send_email_to_team' || actionName === 'send_message')
+        return false;
+      const type = AUTOMATION_ACTION_TYPES.find(
+        action => action.key === actionName
+      ).inputType;
+      if (type === null) return false;
+      return true;
+    },
+    resetAction(index) {
+      this.macro.actions[index].action_params = [];
+    },
+    onDragEnd() {
+      this.dragging = false;
+    },
+    hasError(v) {
+      return !!(v.action_params.$dirty && v.action_params.$error);
+    },
+    async saveMacro() {
+      this.$v.$touch();
+      if (this.$v.$invalid) return;
+      try {
+        const action = this.mode === 'EDIT' ? 'macros/update' : 'macros/create';
+        const successMessage =
+          this.mode === 'EDIT'
+            ? this.$t('MACROS.EDIT.API.SUCCESS_MESSAGE')
+            : this.$t('MACROS.ADD.API.SUCCESS_MESSAGE');
+        let macro = JSON.parse(JSON.stringify(this.macro));
+        macro.actions = actionQueryGenerator(macro.actions);
+        await this.$store.dispatch(action, macro);
+        this.showAlert(this.$t(successMessage));
+        this.$router.push({ name: 'macros_wrapper' });
+      } catch (error) {
+        this.showAlert(this.$t('MACROS.ERROR'));
+      }
+    },
+    formatMacro(macro) {
+      const formattedActions = macro.actions.map(action => {
+        let actionParams = [];
+        if (action.action_params.length) {
+          const inputType = this.macroActionTypes.find(
+            item => item.key === action.action_name
+          ).inputType;
+          if (inputType === 'multi_select') {
+            actionParams = [
+              ...this.getActionDropdownValues(action.action_name),
+            ].filter(item => [...action.action_params].includes(item.id));
+          } else if (inputType === 'team_message') {
+            actionParams = {
+              team_ids: [
+                ...this.getActionDropdownValues(action.action_name),
+              ].filter(item =>
+                [...action.action_params[0].team_ids].includes(item.id)
+              ),
+              message: action.action_params[0].message,
+            };
+          } else actionParams = [...action.action_params];
+        }
+        return {
+          ...action,
+          action_params: actionParams,
+        };
+      });
+      return {
+        ...macro,
+        actions: formattedActions,
+      };
+    },
+    async manifestMacro() {
+      const macroAvailable = this.$store.getters['macros/getMacro'](
+        this.macroId
+      );
+      if (macroAvailable) this.macro = this.formatMacro(macroAvailable);
+      else {
+        const rawMacro = await this.$store.dispatch(
+          'macros/getSingleMacro',
+          this.macroId
+        );
+        this.macro = this.formatMacro(rawMacro);
+      }
+    },
+    fetchMacro() {
+      if (this.macroId) {
```

Comment:
This check is unnecessary as the method is only called when this.$route.params.macroId is present.